src/credentials/CertificationDeclaration.cpp

```diff
@@ -170,6 +183,22 @@ CHIP_ERROR DecodeCertificationElements(const ByteSpan & encodedCertElements, Cer
     }
     VerifyOrReturnError(err == CHIP_END_OF_TLV || err == CHIP_ERROR_UNEXPECTED_TLV_ELEMENT || err == CHIP_NO_ERROR, err);
 
+    if (reader.GetTag() == ContextTag(kTag_AuthorizedPAAList) && reader.GetType() == kTLVType_Array)
+    {
+        ReturnErrorOnFailure(reader.EnterContainer(outerContainer2));
+
+        certElements.AuthorizedPAAListCount = 0;
+        while ((err = reader.Next(AnonymousTag())) == CHIP_NO_ERROR)
+        {
+            ReturnErrorOnFailure(
+                reader.GetBytes(certElements.AuthorizedPAAList[certElements.AuthorizedPAAListCount++], kKeyIdentifierLength));
```

Comment:
A few issues here:

1.  This is not checking that the type is octet string; it will happily read a char string.
2. If the TLV has fewer than `kKeyIdentifierLength` bytes for the string, it will only read whatever it has and leave the rest of the value un-initialized, but later we seem to assume that we have `kKeyIdentifierLength` bytes of value.

Needs tests for the failure cases.
4.

Reply:
fixed. Added more verification checks.

also opened a separate ticket to follow up with testing error cases